### PR TITLE
Xcode 10 changes for test schemes

### PR DESCRIPTION
**Pull request by jkap**
added:
* parallelization
* randomized test order
* whitelisted tests

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Xcode 10 changes for test schemes  
  [jkap](https://github.com/jkap)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: lib/xcodeproj/scheme/test_action.rb [hunk 1]
```diff
@@ -200,6 +230,52 @@ def add_skipped_test(skipped_test)
...
+          @xml_element.elements['SelectedTests'].get_elements('Test').map do |node|
+            TestableReference::SelectedTest.new(node)
+          end
+        end
+
+        # @param [Array<SelectedTest>] tests
+        #         Set the list of SelectedTest this action will run.
+        #
+        def selected_tests=(tests)
+          @xml_element.delete_element('SelectedTests') unless @xml_element.elements['SelectedTests'].nil?
```
[the remark this answers is not in the document]
Comment (jkap) [reply lib/xcodeproj/scheme/test_action.rb]: this is copied from `skipped_tests=` so i removed it there too

File: spec/scheme/test_action_spec.rb [hunk 2]
```diff
@@ -221,6 +221,7 @@ module Xcodeproj
         test_ref.xml_element.name.should == 'TestableReference'
         test_ref.xml_element.attributes.count.should == 1
         test_ref.xml_element.attributes['skipped'].should == 'NO'
+        test_ref.randomized?.should.be.false?
```
[the remark this answers is not in the document]
Comment (jkap) [reply spec/scheme/test_action_spec.rb]: did not know abt this syntax!

**Comment by jkap** [comment]
fixed @segiddins's review in 0a2137e8e09021ee7564530ddabc9d0446ab6a3e

**Pull request merged**